docs/user_guide.md

```diff
@@ -719,6 +721,70 @@ If you are a project admin, you can disable a Robot Account by clicking "Disable
+|Download Helm chart from registry|`CHART PULL`|Repository name, chart name, chart type, chart version, chart size, tag, timestamp of push, username of user who pulled chart|
+|Delete Helm chart from registry|`CHART DELETE`|Repository name, chart name, chart type, chart version, chart size, tag, timestamp of delete, username of user who deleted chart|
+|Image scan completed|`IMAGE SCAN COMPLETED`|Repository namespace name, repository name, tag scanned, image name, number of critical issues, number of major issues, number of minor issues, last scan status, scan completion time timestamp, vulnerability information (CVE ID, description, link to CVE, criticality, URL for any fix), username of user who performed scan|
+|Image scan failed|`IMAGE SCAN FAILED`|Repository namespace name, repository name, tag scanned, image name, error that occurred, username of user who performed scan|
+
+### Webhook Endpoint Recommendations
+
+The endpoint that receives the webhook should ideally meet the following requirements:
+
+- Have a webhook listener that is capable of interpreting the payload and acting upon the information it contains. For example, running a shell script.
+- Support queues and retry configurations, so that failed webhook notifications are queued and retried until the request succeeds or number of retries exceeds the configured maximum.
+
+It is recommended to create a separate endpoint URL for each project. 
+
+### Example Use Cases
+
+You can configure your continuous integration and development infrastructure so that it performs the following types of operations when it receives a webhook notification from Harbor.
+
+- Image push: 
+  - Trigger a new build immediately following a push on selected repositories s or tags.
+  - Notify services or applications that use the image that a new image is available and pull it.
+  - Scan the image using Clair.
+  - Replicate the image to remote registries.
+- Image scanning:
+  - If a vulnerability is found, rescan the image or replicate it to another registry.
+  - If the scan passes, build the image.
+
+### Configure Webhooks
+
+1. Select a project and go to the Webhooks tab.
+  ![Webhooks option](img/webhooks1.png)  
+1. Enter the URL for your webhook endpoint listener.
+  ![Webhook URL](img/webhooks2.png)
+1. If your webhook listener implements authentication, enter the authentication header. 
+1. To implement `HTTPS POST` instead of `HTTP POST`, select the **Verifiy Remote Certficate** check box.
+1. Click **Test Endpoint** to make sure that Harbor can connect to the listener.
+1. Click **Continue** to create the webhook.
+
+When you have created the webhook, you see the status of the different notifications and the timestamp of the last time each notification was triggered. You can click **Disable** to disable notifications.
+
+![Webhook Status](img/webhooks3.png)
+
+**NOTE**: You can only disable and reenable all notifications. You cannot disable and enable selected notifications.
+

```

Comment:
We should also mention that if the webhook fails to send or receives error http response (with non-2xx code) the webhook will be re-sent based on configuration in `harbor.yml`
@wy65701436  @mmpei please confirm

Reply:
yes. and there is a system configuration in `configuration`->`system settings` to enable/disable webhook globally.

Reply:
Added both statements in https://github.com/goharbor/harbor/pull/8869/commits/d86e39bccf0dc1c5fa1260d602ccba47fa405393.